app/src/test/java/com/example/foody/recipe_search/presentation_mvvm/SearchViewModelTest.kt

```diff
@@ -0,0 +1,189 @@
+package com.example.foody.recipe_search.presentation_mvvm
+
+import app.cash.turbine.ReceiveTurbine
+import app.cash.turbine.turbineScope
+import com.example.foody.recipe_search.domain.RecipesSearchRepository
+import com.example.foody.recipe_search.presentation_mvvm.model.RecipeListState
+import com.example.foody.recipe_search.presentation_mvvm.model.SearchScreenState
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.mockk
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.test.StandardTestDispatcher
+import kotlinx.coroutines.test.TestScope
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
+import org.junit.Assert.*
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import java.io.IOException
+
+class SearchViewModelTest {
+    @OptIn(ExperimentalCoroutinesApi::class)
+    private var dispatcher = UnconfinedTestDispatcher()
+    
+    private lateinit var subject: SearchViewModel
+    private lateinit var repoMock: RecipesSearchRepository
+    private lateinit var testSearchTerm: String
+    private val testListOfRecipes = getRecipeListTestData(5)
+    
+    @OptIn(ExperimentalCoroutinesApi::class)
+    @Before
+    fun setUp() {
+        Dispatchers.setMain(dispatcher)
+        repoMock = mockk()
+    }
+    
+    @OptIn(ExperimentalCoroutinesApi::class)
+    @After
+    fun tearDown() {
+        Dispatchers.resetMain()
+    }
+    
+    @Test
+    fun `Given we have a valid search term, Then list of recipes is fetched`() {
+        // Arrangement
+        testSearchTerm = "pie"
+        coEvery { repoMock.search(testSearchTerm) } returns testListOfRecipes
+        createViewModel()
+        
+        // Action
+        subject.updateSearchTerm(testSearchTerm)
+        
+        // Assertion, Testing that searchTerm is set to input value
+        assertEquals(testSearchTerm, subject.state.value.searchBarState.searchTerm)
+        
+        // Action
+        subject.search()
+        
+        // Assertions
+        // Testing that after calling a search function with valid searchTerm, the state is Success
+        coVerify { repoMock.search(testSearchTerm) }
+        assertEquals(RecipeListState.Success(testListOfRecipes), subject.state.value.recipeListState)
+        
+        // testing that searchTerm is set to empty after the search
+        assertEquals("", subject.state.value.searchBarState.searchTerm)
+        // testing that the searchBar is hidden
+        assertEquals(false, subject.state.value.searchBarState.expandedState)
+    }
+    
+    @Test
+    fun `Given we have an empty search term (or no results), Then the state is Empty`() {
```

Comment:
This tests what happens if the search returns an empty list, but has nothing to do with an empty string. 
Even if search term is something else, and coEvery for that search returns empty list, the result will be the same.
Also, if search term is empty string, and coEvery call result is set to something other than empty list, than the result will not be `Empty`, no matter the search term was empty string.
```suggestion
    fun `Given we have an empty search term (or no results), Then the state is Empty`() {
```

Reply:
Ok, but then I need another way to test that

Reply:
Even maybe to think of that as functionality. What should happen when user clicks on search if the input field is empty.

Reply:
Ok. Still left to do ...